src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,334 @@
+        if (!is_dir("$plugins_dir")) {
+            mkdir("$plugins_dir", 0755, true);
+        }
+        return $plugins_dir . $plugin;
+    }
+
+    /**
+     * Get the plugin installation method.
+     *
+     * @param string Plugin name
+     * @return string Plugin installation method
+     */
+    protected function getInstallMethod($plugin)
+    {
+        $slash = $this->getSlash();
+        $plugin_dir = $this->getPluginDir($plugin);
+        $git_dir = $plugin_dir . $slash . '.git';
+        if (is_dir("$git_dir") && $this->commandExists('git')) {
+            return 'git';
+        }
+        $composer_json = $plugin_dir . $slash . 'composer.json';
+        if (file_exists($composer_json)) {
+            return $this->commandExists('composer') ? 'composer' : 'archive';
+        }
+        return 'unknown';
+    }
+
+    /**
+     * Update a specific plugin.
+     *
+     * @param string $plugin Plugin name
+     */
+    protected function updatePlugin($plugin)
+    {
+        $plugins_dir = $this->getPluginDir();
+        $plugin_dir = $plugins_dir . $plugin;
+        if (!is_dir("$plugin_dir")) {
+            $message = "{$plugin} is not installed.";
+            throw new TerminusNotFoundException($message);
+        }
+        $messages = array();
+        $message = "Updating {$plugin}...";
+        $this->log()->notice($message);
+        $method = $this->getInstallMethod($plugin);
+        switch ($method) {
+            case 'git':
+                $installed_version = $this->getInstalledVersion($plugin_dir);
+                $latest_version = $this->getLatestVersion($plugin_dir);
+                if ($installed_version < $latest_version) {
+                    exec("cd \"$plugin_dir\" && git checkout {$latest_version}", $messages);
+                } else {
+                    $messages[] = "Already up-to-date.";
+                }
+                break;
+
+            case 'composer':
+                $composer_info = $this->getComposerInfo($plugin);
+                $project = $composer_info['name'];
+                $packagist_url = "https://packagist.org/packages/{$project}";
+                if ($this->isValidUrl($packagist_url)) {
+                    exec("rm -rf \"{$plugin_dir}\" && composer create-project --prefer-source --keep-vcs -n -d {$plugins_dir} {$project}:~1", $messages);
+                } else {
+                    $messages[] = "{$packagist_url} is not a valid Packagist project.";
+                }
+                break;
+
+            case 'archive':
+            default:
+                if ($this->commandExists('curl') && $this->commandExists('tar')) {
+                    $project = 'unknown';
+                    $composer_info = $this->getComposerInfo($plugin);
+                    if (!empty($composer_info)) {
+                        $project = $composer_info['name'];
+                    }
+                    $archive_url = "https://github.com/{$project}/archive/1.x.tar.gz";
```

Comment:
Right, but there is no way of really knowing for sure since we don't store that anywhere after an install.  If it can't be found via github.com, then it will just send a message that the project doesn't exist.